src/utils/contract.ts

```diff
@@ -99,9 +99,9 @@ export default class Contract extends BaseConnection {
         to: this.address,
       },
       'latest',
-    ];
-    if (this.web3Provider) {
-      return await this.web3Provider
+    ] as const;
+    if (this.provider) {
+      return await this.provider
         .sendAsync('eth_call', params)
         .then(resp => resp.json());
```

Comment:
Web3 provider responses have different API (they don't have it even, as they come as parsed jsons). So this doesn't work unfortunately

Reply:
I have added an if there in case if the provider is custom as the one we implemented in previous tests 
```const provider = {
        sendAsync: (method: string, params: any) => {
          return nodeFetch(secretInfuraLink(), {
            method: 'POST',
            body: JSON.stringify({
              method,
              params,
              jsonrpc: '2.0',
              id: 1,
            }),
          });
        },
      };```

and after removing all the mocks internally it passes everything again.